### DISCUSSION TITLE
How does RandomNodeSplit and RandomLinkSplit work together?

### DISCUSSION THREAD

**Discussion by hash-ir** [pyg-team/pytorch_geometric#3839] [Q&A] [answered]
Hi, 

I have created train, val, and test masks using `RandomNodeSplit`. Then, I create the associated loaders using `NeighborLoader`. Since I am using autoencoder architectures, I need to split the edges into _pos_ and _neg_ edge indices. I found the `RandomLinkSplit` but I don't know how it works with the node split already done. Do I apply this transformation on the full dataset before the node split or to the train, val, and test batches separately during training? In the latter case, it would again divide, for example, a training batch, into _pos_ and _neg_ train, val, and test edge indices that do not make sense.

Basically, I want to avoid a data leak in terms of the node or edge indices between the train, val, and test sets. 

Thanks,
Hashir

**Comment by rusty1s** [comment, marked as answer]
Can you say a bit more about what it means to apply `RandomNodeSplit` and `RandomLinkSplit` together? What do you want to try to achieve? What do you expect to happen?

One thing I could think of is to split the graph into 3 subgraphs via `data.subgraph()` based on the node-level masks coming from `RandomNodeSplit`. Each subgraph can then be further splitted via `RandomLinkSplit`.

**Reply by hash-ir** [reply]
I want to split the dataset based on nodes and get positive and negative edge labels and indices for the autoencoder. What I notice is that `RandomLinkSplit` also divides the `data` object into train, val, and test. This is where I am a bit confused.

If I split the graph using `data.subgraph()`, the resulting subgraphs will then again be divided into three sets via `RandomLinkSplit`.

**Reply by rusty1s** [reply]
Yeah, you can probably omit the use of `RandomLinkSplit`, and treat the isolated subgraphs obtained from `RandomNodeSplit` as training, validation and test graphs for link prediction.
